services/hooks/src/data.js

```diff
@@ -221,5 +221,42 @@ LastFire.prototype.definition = function() {
   });
 };
 
+LastFire.expires = async function(Hook, now, n = 100) {
+  const hookIds = new Set();
+  let count = 0;
+  await Hook.scan({},
+    {
+      handler: async hook => {
+        const { hookId } = await hook.definition();
```

Comment:
you'll need to get the hookGroupId as well, and use that in the `this.scan` below, to ensure that hooks with different hookGroupIds but the same hookId don't get confused.

Reply:
hookId is unique how there would be confusion? Later in line 243 I extracted the hookGroupId as well before removing the entity. Yeah but maybe using hookGroupId will lead to faster scan in the LastFires table. Would using `query` instead of `scan` in LastFires make it faster as in somewhere it says avoids full table scan.

Reply:
ah i yeah got it.